sway-core/src/semantic_analysis/namespace/trait_map.rs

```diff
@@ -191,8 +191,14 @@ impl TraitMap {
                     second_impl_span: impl_span.clone(),
                 });
             } else if types_are_subset {
+                let type_args_are_equal = trait_type_args.len() == map_trait_type_args.len()
```

Comment:
This check should be encapsulated by the `types_are_subset` check, so you should need to add this specifically. If its not, then there may be a bug with that check.

Reply:
This check is encapsulated by the `traits_are_subset` but not by `types_are_subset`.

The most recent commit replaces `type_args_are_equal` usage by `!traits_are_subset && !is_impl_self`.